scalding-spark/src/main/scala/com/twitter/scalding/spark_backend/Op.scala

```diff
@@ -144,18 +145,35 @@ object Op extends Serializable {
         input.run(session).map { rdd => fn(widen(rdd)) })
   }
 
-  final case class Merged[A](left: Op[A], right: Op[A]) extends Op[A] {
+  final case class Merged[A](pc: PartitionComputer, left: Op[A], tail: List[Op[A]]) extends Op[A] {
     @transient private val cache = new FutureCache[SparkSession, RDD[_ <: A]]
 
     def run(session: SparkSession)(implicit ec: ExecutionContext): Future[RDD[_ <: A]] =
       cache.getOrElseUpdate(session, {
         // start running in parallel
         val lrdd = left.run(session)
-        val rrdd = right.run(session)
-        for {
-          l <- lrdd
-          r <- rrdd
-        } yield widen[A](l) ++ widen[A](r)
+        val tailRunning = tail.map(_.run(session))
+        tailRunning match {
+          case Nil => lrdd
+          case nonEmpty =>
+            // start all the upstream in parallel:
+            val rrdd = Future.traverse(nonEmpty) { frdd => frdd }
+            for {
+              l <- lrdd
+              rs <- rrdd
+            } yield {
+              val rdds = l :: rs
+              val rddAs = rdds.map(widen[A](_))
+              val merged = new UnionRDD(session.sparkContext, rddAs)
+              val partitions = merged.getNumPartitions
+              val newPartitions = pc(partitions)
+              if (partitions != newPartitions) {
```

Comment:
should this be `if (partitions > newPartitions)`? i'm intending to make this change as well.

Reply:
I'll just merge your change, but I think `newPartitions < partitions` reads better.